x-pack/legacy/plugins/ml/public/jobs/new_job_new/recognize/components/edit_job.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ModuleJobUI } from '../page';
+import { usePartialState } from '../../../../components/custom_hooks';
+import { composeValidators, maxLengthValidator } from '../../../../../common/util/validators';
+import { isJobIdValid } from '../../../../../common/util/job_utils';
+import { JOB_ID_MAX_LENGTH } from '../../../../../common/constants/validation';
+import { JobGroupsInput } from '../../common/components';
+import { JobOverride } from '../../../../../common/types/modules';
+
+interface EditJobProps {
+  job: ModuleJobUI;
+  jobOverride: JobOverride | undefined;
+  existingGroupIds: string[];
+  onClose: (job: JobOverride | null) => void;
+}
+
+/**
+ * Edit job flyout for overriding job configuration.
+ */
+export const EditJob: FC<EditJobProps> = ({ job, jobOverride, existingGroupIds, onClose }) => {
+  const [formState, setFormState] = usePartialState({
+    jobGroups: (jobOverride && jobOverride.groups) || job.config.groups,
+  });
+  const [validationResult, setValidationResult] = useState<Record<string, any>>({});
+
+  const groupValidator = composeValidators(
+    (value: string) => (isJobIdValid(value) ? null : { pattern: true }),
+    maxLengthValidator(JOB_ID_MAX_LENGTH)
+  );
+
+  const handleValidation = () => {
+    const jobGroupsValidationResult = formState.jobGroups
+      .map(group => groupValidator(group))
+      .filter(result => result !== null);
+
+    setValidationResult({
+      jobGroups: jobGroupsValidationResult,
+      formValid: jobGroupsValidationResult.length === 0,
+    });
+  };
+
+  useEffect(() => {
+    handleValidation();
+  }, [formState.jobGroups]);
+
+  const onSave = () => {
```

Comment:
If I remove all the groups on a job, and then save, and create the jobs, the job ends up with the groups defined in the module. I think it should end up with no groups in this case, unless there is a good reason to fall back to the groups defined in the module?

Reply:
In request payload, I send `groups: []`, so it looks like API ignores empty array and fallback to default module groups.

Reply:
the problem here is with the use of lodash's `merge` being used for combining the override settings and the job.
it works correctly for overriding objects, but not for arrays.

Reply:
there is no `merge` used for the `jobOverrides`, only for kibana objects response, which is not related